File: PlazaAudioVisualServer/src/main/java/org/example/model/Media.java
package org.example.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.UUID;

@Entity
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "media")
@ToString
public class Media {
    private String id;
    private Long size;
    private String type;
    private String name;
    private String description;
    private int dimension;
    @ToString.Exclude
    private User user;

    public Media(Long size, String type, String name, String description, int dimension) {
        this.id = UUID.randomUUID().toString();
        this.size = size;
        this.type = type;
        this.name = name;
        this.description = description;
        this.dimension = dimension;
    }

    @Id
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Long getSize() {
        return size;
    }

    public void setSize(Long size) {
        this.size = size;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getDimension() {
        return dimension;
    }

    public void setDimension(int dimension) {
        this.dimension = dimension;
    }

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: plaza_audiovisual_interfaz/lib/model/media.dart
import 'dart:ffi';

import 'package:get/get.dart';
import 'package:plaza_audiovisual_interfaz/model/user.dart';

class Media {
  RxString? id = "".obs;
  int? size;
  RxString? type = "".obs;
  RxString? name = "".obs;
  RxString? description = "".obs;
  int? dimension;

  Media({
    this.id,
    this.size,
    this.type,
    this.name,
    this.description,
    this.dimension,
  });

  Media.fromJson(Map<String, dynamic> json) {
    id!.value = json['id'];
    size = json['size'];
    type!.value = json['type'];
    name!.value = json['name'];
    description!.value = json['description'];
    dimension = json['dimension'];
  }
}